src/Fixer/ControlStructure/NoBreakCommentFixer.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ControlStructure;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Fixer for rule defined in PSR2 ¶5.2.
+ */
+final class NoBreakCommentFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'There must be a comment when fall-through is intentional in a non-empty case body.',
+            [
+                new CodeSample("<?php\nswitch (\$foo)\n{    case 1:\n        foo();\n    case 2:\n        bar();\n}"),
+                new CodeSample("<?php\nswitch (\$foo)\n{    case 1:\n        foo();\n        // no break\n        break;\n    case 2:\n        bar();\n}"),
+            ],
+            'Adds a "no break" comment before fall-through cases, and removes it if there is no fall-through.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_CASE, T_DEFAULT]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $position => $token) {
+            if ($token->isGivenKind([T_CASE, T_DEFAULT])) {
+                $this->fixCase($tokens, $position);
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $casePosition
+     */
+    private function fixCase(Tokens $tokens, $casePosition)
+    {
+        $empty = true;
+        $fallThrough = true;
+        $commentPosition = null;
+        for ($i = $tokens->getNextTokenOfKind($casePosition, [':', ';']) + 1, $max = count($tokens); $i < $max; ++$i) {
+            if ($tokens[$i]->isGivenKind([T_SWITCH, T_IF, T_ELSE, T_ELSEIF, T_FOR, T_FOREACH, T_WHILE, T_DO, T_FUNCTION, T_CLASS])) {
```

Comment:
not sure...
array packing (https://blog.blackfire.io/php-7-performance-improvements-packed-arrays.html) + immutable array (https://blog.blackfire.io/php-7-performance-improvements-immutable-arrays.html) would work better than static array IMO (as it would register some static memory that would not be freezed until end of whole tool execution)

I would leave it as-is